client/src/modules/reports/generate/inventory_file/inventory_file.config.js

```diff
@@ -0,0 +1,89 @@
+angular.module('bhima.controllers')
+  .controller('inventory_fileController', InventoryFileConfigController);
+
+InventoryFileConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state',
+  'LanguageService', 'moment',
+];
+
+function InventoryFileConfigController($sce, Notify, SavedReports, AppCache, reportData, $state,
+  Languages, moment) {
+  var vm = this;
+  var cache = new AppCache('configure_inventory_file_report');
+  var reportUrl = 'reports/stock/inventory';
+
+  vm.previewGenerated = false;
+
+  vm.dateTo = new Date();
+
+  // chech cached configuration
+  checkCachedConfiguration();
+
+  vm.setupDepot = function setupDepot(depot) {
+    vm.depot = depot;
+  };
+
+  vm.setupInventory = function setupInventory(inventory) {
+    vm.inventory = inventory;
+  };
+
+  vm.clear = function clear(key) {
+    delete vm[key];
+  };
+
+  vm.clearPreview = function clearPreview() {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.preview = function preview(form) {
+    var options;
+    var params;
+
+    if (form.$invalid) { return 0; }
+
+    params = {
+      depot_uuid : vm.depot.uuid,
+      inventory_uuid : vm.inventory.uuid,
+      dateTo : vm.dateTo,
+    };
+
+     // update cached configuration
+    cache.reportDetails = angular.copy(params);
+
+    // format date for the server
+    params.dateTo = moment(params.dateTo).format('YYYY-MM-DD');
```

Comment:
In the future, if we are going to stringify our dates before sending them to the server, we should do it all in the same spot.  `$httpParamSerializer` is the best place to detect if the parameter is a date and then stringify it appropriately.

This should be discussed first though.